app/components/primer/alpha/dialog/header.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module Primer
+  module Alpha
+    class Dialog
+      # A `Dialog::Header` is a compositional component, used to render the
+      # Header of a dialog. See <%= link_to_component(Primer::Alpha::Dialog) %>.
+      class Header < Primer::Component
+        # @param title [String] The title of the dialog.
+        # @param subtitle [String] The subtitle of the dialog. This will also set the `aria-describedby` attribute.
+        # @param hide_divider [Boolean] If true the visual dividing line between the body and footer will be hidden
+        # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+        def initialize(
+          id:,
+          title:,
+          subtitle: nil,
+          subtitle_id: nil,
+          hide_divider: false,
```

Comment:
Since this should be false by default, I wonder if we could rename it to `show_divider`. This prop name also aligns a bit better with how PRC has referenced "divider" props.

```suggestion
          show_divider: false,
```